fix(coderd/prometheusmetrics)!: filter deleted wsbuilds to reduce db load

mafredri opened the pull request:
This change removes the `GetLatestWorkspaceBuilds` query which includes
all workspaces for all time (including deleted). This allows us to also
stop using `GetProvisionerJobsByIDs` for said builds as the job status
is included in `GetWorkspaces` called separately.

**BREAKING CHANGE**: The `coderd_api_workspace_latest_build` Prometheus
metric no longer includes builds belonging to deleted workspaces, as
such, this metric will show fewer statuses.

Fixes coder/internal#717

mafredri committed:
fix(coderd/prometheusmetrics): no deleted wsbuilds to reduce db load

This change removes the `GetLatestWorkspaceBuilds` query which includes
all workspaces for all time (including deleted). This allows us to also
stop using `GetProvisionerJobsByIDs` for said builds as the job status
is included in `GetWorkspaces` called separately.

Fixes coder/internal#717

mafredri commented on coderd/prometheusmetrics/prometheusmetrics_test.go:
```diff
@@ -323,6 +349,33 @@ func TestWorkspaceLatestBuildStatuses(t *testing.T) {
+			require.NoError(t, err)
+			for range 1 + n {
+				insertDeleted(t, db, u, org)
+			}
```
Review: This fails in the old implementation, but I thought it best to formalize the assumption. Considering the old implementation did not test for this, it seems like validation for the new approach.

johnstcn reviewed (approved):
This change makes sense to me.

dannykopping reviewed (approved):
Seems reasonable 👍 

Please ensure the semantics of this change are documented in the release notes for the next release.
We could clarify the definition of this metric too, which may help operators:

https://coder.com/docs/admin/integrations/prometheus#available-metrics

Currently it's quite brief: "The latest workspace builds with a status."

mafredri committed:
update help for metrics

mafredri commented:
I've updated the PR to be a "breaking" change as well as describe the changed behavior. I've also updated the helps for the metrics to mention that it's for non-deleted workspaces.

mafredri committed:
Merge branch 'main' into mafredri/fix-prommetrics-db-load
